lib/Autocomplete.js

```diff
@@ -7,20 +7,106 @@ let _debugStates = []
 let Autocomplete = React.createClass({
 
   propTypes: {
+    /**
+     * The items to display in the dropdown menu
+     */
+    items: React.PropTypes.array.isRequired,
+    /**
+     * The value to display in the input field
+     */
     value: React.PropTypes.any,
+    /**
+     * Arguments: `event: Event, value: String`
+     *
+     * Invoked every time the user changes the input's value.
+     */
     onChange: React.PropTypes.func,
+    /**
+     * Arguments: `value: String, item: Any`
+     *
+     * Invoked when the user selects an item from the dropdown menu.
+     */
     onSelect: React.PropTypes.func,
+    /**
+     * Arguments: `item: Any, value: String`
+     *
+     * Invoked for each entry in `items` and its return value is used to
+     * determine whether or not it should be displayed in the dropdown menu.
+     * By default all items are always rendered.
```

Comment:
Yes. There's no real reason to have `props.shouldItemRender` any longer. Now that `Autocomplete` is a managed component you need to keep track of `props.value`, so there's no reason why you can't just filter `props.items` before passing it in. Before, when we kept `value` in state, there was no guarantee that the consumer would be tracking `value`, so it made more sense to provide this convenience API so you could get more done "within" the component itself. I think we'll remove this prop for v2.